fix: Method _should_use_plan only returns true for local sqlite provider

nossrannug opened the pull request:
Only the SQLite local provider supports _plan and _apply_diffs. Updated
the _should_use_plan method to only return true for that setup so that other local providers don't need to implement these methods right now.

Signed-off-by: Gunnar Sv Sigurbjörnsson <gunnar.sigurbjornsson@gmail.com>

**What this PR does / why we need it**:
Only the SQLite provider supports _plan and _apply_diffs today. Before this can be added to other providers there needs to be some discussion around the InfraObject and how the update and teardown is performed.

My main worry at the moment is that the InfraObject can be constructed from a proto and then the update method can be called. This requires all the connection information and credentials to be present in the infra object proto.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

felixwang9817 reviewed (approved):
/lgtm

felixwang9817 commented:
oops missed a small bug: looks like you need to do a null check on `online_store` here:
```
self.config.online_store.type == "sqlite"
```
integration tests are failing

separately, we recently moved to [semantic release](https://semantic-release.gitbook.io/semantic-release/), so all PR titles now need to prefixed with e.g. "refactor" or "fix" depending on the type of PR - you can see some other PRs for an example, and the set of possible prefixes is documented [here](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#type)

nossrannug commented:
> @nossrannug oops missed a small bug: looks like you need to do a null check on `online_store` here:
> 
> ```
> self.config.online_store.type == "sqlite"
> ```
> 
> integration tests are failing
> 
> separately, we recently moved to [semantic release](https://semantic-release.gitbook.io/semantic-release/), so all PR titles now need to prefixed with e.g. "refactor" or "fix" depending on the type of PR - you can see some other PRs for an example, and the set of possible prefixes is documented [here](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#type)

Thanks @felixwang9817, I'm sorry I missed that. I've added the null check and updated the title.

felixwang9817 reviewed (approved):
/lgtm

woop commented:
:tada: This PR is included in version 0.19.0 :tada:

The release is available on [GitHub release](https://github.com/feast-dev/feast/releases/tag/v0.19.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
